components/planner/src/dynamo/planner/utils/planner_core.py

```diff
@@ -137,7 +155,17 @@ async def get_workers_info(self):
             raise RuntimeError(f"Failed to get decode worker endpoints: {e}")
         return p_endpoints, d_endpoints
 
-    def observe_metrics(self):
+    async def observe_metrics(self):
+        self.p_endpoints, self.d_endpoints = await self.get_workers_info()
+        logger.info(
```

Comment:
```suggestion
        logger.debug(
```

This will flood the logs correct given the prometheus scrape interval. If this is not the intended behavior, lets change to debug